pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -354,7 +358,7 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 	}
 
 	// some of the volume additions may have failed, should not mark this pod as fully processed
-	if allVolumesAdded {
+	if volumeAdded || allVolumesAdded {
```

Comment:
Add a comment here explaining why if a single volume is added it's ok, because it goes against the current comment.

Reply:
Also mention in the comment that this is only going to handle the scenario where the pod was successfully processed once, and errors on a reprocessing.

The scenario where the initial pod handling fails and we retry frequently is not going to be solved by this pr

Reply:
Comment has been added.